### ISSUE TITLE
Issue Parsing x-www-form-urlencoded POST Requests

### ISSUE THREAD

**Issue by antsankov** [Workiva/go-rest#50]
I was working on an endpoint that needed to receive requests in the format x-www-form-urlencoded. This causes issues with the JSON decoder line in [handler.go](https://github.com/Workiva/go-rest/blob/master/rest/handler.go#L138-L141):
```
data, err := decodePayload(payloadString(r.Body))
		if err != nil {
			// Payload decoding failed.
			ctx = ctx.setError(BadRequest(err.Error()))
		}
```
Obviously if the data isn't in JSON format it can't be decoded. However if you even attempt to decode r.Body, because it is a io stream, it is emptied and can't be accessed. 

I devised a way of fixing this in my project, and was curious if there was any interest in a pull request for this issue.

**Comment by tylertreat-wf** [comment]
Yeah, ideally payload decoding could be pluggable based on the Content-Type. Feel free to open a PR.
